### ISSUE TITLE
RabbitMQ reject doesn't work on consumer exception

### ISSUE THREAD

**Issue by madonzyaccenture** [magento/magento2#32670]
### Preconditions (*)
1. Magento 2.3.3

### Steps to reproduce (*)
1. Run any bulk WEBAPI import
2. Start the consumer
3. Within consumer throw an exception

### Expected result (*)
1. Message should returned to the queue

### Actual result (*)
1. Message was not returned to the queue, thus lost

Here is a code that explains that if an exception was thrown, then reject the message: vendor/magento/module-asynchronous-operations/Model/MassConsumerEnvelopeCallback.php:120
![image](https://user-images.githubusercontent.com/51681547/122412703-6bcabe00-cfa3-11eb-9692-1d34c2671596.png)
I found that requeue argument is set to false, so I manually changed it to true for test purposes. The message was not requeued as well.

If that is expected behavior, please tell me how can I manipulate basic_ack and basic_nack/basic_reject within consumers? I also posted the question on StackOverflow: https://magento.stackexchange.com/questions/335005/magento-2-rabbitmq-ack-or-reject-message/335023#335023

**Note:** Either I set requeue as **true** or **false** the behaviour is same. Exceptions, don't get requeue whether requeue variable is set as true. Please Fix the implementation of **requeue** argument under **reject** method

**Comment by engcom-Golf** [comment]
@madonzyaccenture As suggested rabbitMQ is configured, Product asynchronous bulk API **/rest/async/bulk/V1/products** is called.
Consumer **async.operations.all** queue is processed successfully without any exception. 

Please note that MessageLockException, ConnectionLostException, NotFoundException exceptions are already handled. 

Can you please assist me to reproduce the 3rd Step. Your exception occur at what time?

**Comment by madonzyaccenture** [comment]
@engcom-Golf just throw an \Exception and see if the message is removed from the queue. Actual behavior is that message is removed, but it's an issue as message was not proceed properly.

**Comment by engcom-Golf** [comment]
@madonzyaccenture This is not a issue, It is an expected behaviour for all the Exceptions. If any Exception occur, It will be removed from Queue, Also proper Error Message will be shown in the Magento Admin panel - **Bulk Log**

Let me Explain you with Example.
1. I tried to insert 2 Products, having different SKUs but same URL KEY via Bulk Async API
![image](https://user-images.githubusercontent.com/51681547/122062280-e7decd80-ce0c-11eb-9553-feb47f103b5a.png)
![image](https://user-images.githubusercontent.com/51681547/122062371-faf19d80-ce0c-11eb-9e82-e0767e0dc9cf.png)
![image](https://user-images.githubusercontent.com/51681547/122062444-0ba21380-ce0d-11eb-8dbc-61ce9530204d.png)
![image](https://user-images.githubusercontent.com/51681547/122062610-312f1d00-ce0d-11eb-8c98-f044e3a63259.png)
![image](https://user-images.githubusercontent.com/51681547/122062718-499f3780-ce0d-11eb-8da6-5973ce038f9f.png)

**Comment by madonzyaccenture** [comment]
@engcom-Golf nope, imagine that issue occurs not because of duplicated entity, but because we have some 3D party API that must be called before save, but currently is not responding, thus entity could not be saved. Developer need to decide to remove or requeue, currently we don't have such option. Messages will always be deleted.

**Comment by engcom-Golf** [comment]
@madonzyaccenture Thank you so much for detailed description.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sdzhepa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-42769` were created

**Issue Available**: @sdzhepa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by nwcasebolt** [comment]
+1 to this aging request for a fix. I encountered the same behavior today while testing locally on 2.4.7-p5. The option to requeue rejected messages is essential.

**Comment by engcom-Bravo** [comment]
Hi @madonzyaccenture,

Thanks for your reporting and collaboration.

We have tried to reproduce the issue in Latest 2.4-develop instance and the issue is no more reproducible.we are closing this issue.Please feel free to reopen the issue if the issue still persists.

Thanks.
